pkg/controller/utils/velero.go

```diff
@@ -0,0 +1,42 @@
+package utils
+
+import (
+	"context"
+	"fmt"
+
+	velerov1 "github.com/heptio/velero/pkg/apis/velero/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	// DefaultExcludedBackupResources is the deault list of excludes when backing up resources.
+	DefaultExcludedBackupResources = []string{
+		// NOTE: even though pods is in the "core" group, specifying "pods.core" here will cause
+		// pods to BE BACKED UP (not excluded)!!!
+		"pods",
+		"jobs.batch",
+	}
+)
+
+// CreateVeleroBackupObjectForNamespace creates a Velero Backup object for the namespace specified.
+// The Backup options are set specifically for Hive object backups.
+// DO NOT use this function call for any other type objects as it may not back them up correctly.
+func CreateVeleroBackupObjectForNamespace(client client.Client, namespace string) error {
+	backupName := fmt.Sprintf("backup-%v-%v", namespace, uuid.NewUUID())
```

Comment:
We can definitely do a datestamp instead of a UUID. Worst case scenario would be when we go create the velero backup object we'd get an error. We can then return a reconcile result with a delay / requeue.

Is that what we should do?

Reply:
don't know if the retry code is worth it just to be able to have timestamped names. maybe just going with the UUID is enough, and some queries with piping to jq can sort out each backup objects creation date fields.

@staebler WDYT?

Reply:
I like timestamp better. I think it makes it easier for an end user to deal with if they are sortable by name.

I also think it is worthwhile to place limiting on how rapidly backups are created. I don't know if it is helpful to create backups so rapidly that two backups may have the same timestamp. For what it's worth, having a backup resource would help with that.

Reply:
+1 for timestamp in the name.